multimodal/vl2l/src/mlperf_inference_multimodal_vl2l/evaluation.py

```diff
@@ -0,0 +1,184 @@
+        if pred_cat == true_cat:
+            intersection_count += 1
+        else:
+            # Stop as soon as a mismatch is found (enforces hierarchical match)
+            break
+
+    pred_length = len(predicted_categories)
+    true_length = len(true_categories)
+
+    return intersection_count, pred_length, true_length
+
+
+def calculate_hierarchical_metrics(data: list[tuple[str, str]]) -> float:
+    """Calculates the aggregate hP, hR, and hF scores for a list of samples.
+
+    Args:
+        data: A list of tuples, where each tuple is
+            (predicted_path_str, true_path_str).
+
+    Returns:
+        F1 score
+    """
+    total_intersection = 0
+    total_predicted_length = 0
+    total_true_length = 0
+
+    # 1. Aggregate the components across all samples
+    for pred_path, true_path in data:
+        intersection, pred_len, true_len = \
+            get_hierarchical_components(pred_path, true_path)
+
+        total_intersection += intersection
+        total_predicted_length += pred_len
+        total_true_length += true_len
+
+    # 2. Calculate hP and hR
+    hp = total_intersection / total_predicted_length \
+        if total_predicted_length > 0 else 0.0
+    hr = total_intersection / total_true_length \
+        if total_true_length > 0 else 0.0
+
+    return 0.0 if hp + hr == 0 else 2 * (hp * hr) / (hp + hr)
+
+
+def calculate_exact_match(generated_text: str, original_text: str) -> float:
+    """Calculates binary Exact Match (EM) score.
+
+    We clean the text (lowercase, strip whitespace) for a fairer comparison.
+
+    Args:
+        generated_text: Output from the VLM.
+        original_text: Ground truth information from the dataset.
+
+    Returns:
+        1 if the values match or 0 otherwise
+    """
+    gen = generated_text.strip().lower()
+    orig = original_text.strip().lower()
+
+    return 1.0 if gen == orig else 0.0
+
+
+def alt_f1_score(data: list[tuple[str, str]]) -> float:
+    """Alt method to calculate hierarchical F1."""
+    y_pred_raw = []
+    y_true_raw = []
+
+    for pred, src in data:
+        path1 = pred.split(" > ")
+        path2 = src.split(" > ")
+
+        y_pred_raw.append(path1)
+        y_true_raw.append(path2)
+
+    # 2. Find the global maximum length across ALL samples
+    # We check the longest path in both true and pred lists
+    max_len = max(len(p) for p in y_true_raw + y_pred_raw)
+
+    # 3. Pad all lists to the global max_len
+    for i in range(len(y_true_raw)):
+        # Pad Truth
+        pad_len_true = max_len - len(y_true_raw[i])
+        y_true_raw[i] += [""] * pad_len_true
+
+        # Pad Prediction
+        pad_len_pred = max_len - len(y_pred_raw[i])
+        y_pred_raw[i] += [""] * pad_len_pred
+
+    # 4. Convert to numpy arrays
+    y_true = np.array(y_true_raw)
+    y_pred = np.array(y_pred_raw)
+
+    # 5. Calculate Score
+    return f1(y_true, y_pred)
+
+
+def run_evaluation(filename: FilePath, dataset: "DatasetCLI") -> None:
```

Comment:
```suggestion
def run_evaluation(filename: FilePath, dataset: DatasetCLI) -> None:
```
I remember that this was a problem in python <= 3.8 or some earlier version, but later version of python all adopted delayed type evaluation, thereby you won't need to quote the type. In earlier python version, you can get it supported via:
```python
from __future__ import annotations
```